### PR TITLE
Horner docs

### LINKED ISSUE #1140: Doc: Horner transformation is not documented

### PR DESCRIPTION
**Pull request by kbevers**
Here's a first go at adding proper documentation for the horner transformation. It's mostly based on EPSG GN 7-2 but using different symbology that matches the parameters better. The code is extremely difficult to make sense of (well, at least the core part) so I may have misrepresented the math. I think it is okay but an extra pair of eyes wouldn't hurt.

I also moved the parameter include rst-files outside of the Sphinx source dir since they produce warnings that (at least locally for me) break the doc build.

Closes #1140

### CODE REVIEW

**Comment by rouault** [comment]
Hum, perhaps you should try to upgrade your sphynx version ? I don't get warnings with 
```
$ sphinx-build --version
sphinx-build 1.8.1
```

**Comment by kbevers** [comment]
That may sort the problem, actually. This was on my work computer which is generally not very up to date... The warnings has been annoying me for a long time though, I have just not had a reason to do something about them before now. I don't mind reverting it if you think it is better to have the includes closer to where they are needed.

**Comment by rouault** [comment]
> I don't mind reverting it if you think it is better to have the includes closer to where they are needed.

yes, the current situation seems more logical to me. A 'pip install --upgrade sphinx' should probably fix your wrnings

File: docs/source/operations/transformations/horner.rst [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+frames where one or both are inhomogenous or internally distorted. This will typically
+be reference frames created before the introduction of space geodetic techniques such
+as GPS.
+
+Horner polynomials, or Multiple Regression Equations as they are also known as, have
+their strenght in being able to create fairly complicated mappings between coordinate
+reference frames while still being lightweight in both computational cost and disk
+space used.
+
+PROJ implements two versions of the Horner evalation scheme: Real and complex
```
Comment (mwtoews) [inline docs/source/operations/transformations/horner.rst]: evalution -> evaluation

File: docs/source/operations/transformations/horner.rst [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+| **Output type** | Geodetic and projected coordinates                                |
++-----------------+-------------------------------------------------------------------+
+
+The Horner polynomial evalution scheme is used for transformations between reference
+frames where one or both are inhomogenous or internally distorted. This will typically
+be reference frames created before the introduction of space geodetic techniques such
+as GPS.
+
+Horner polynomials, or Multiple Regression Equations as they are also known as, have
+their strenght in being able to create fairly complicated mappings between coordinate
```
Comment (mwtoews) [inline docs/source/operations/transformations/horner.rst]: strenght -> strength
<s>fairly</s> complicated

File: docs/source/operations/transformations/horner.rst [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+Required
+-------------------------------------------------------------------------------
+
+Below is a list of required parameters that can be set for the Horner polynomial
+transformation. As stated above, the transformation takes to forms, either using
+real or complex polynomia. These are divided into separate sections below.
+Parameters from the two sections are mutually exclusive, that is parameters
+describing real and complex polynomia can't be mixed.
```
Comment (mwtoews) [inline docs/source/operations/transformations/horner.rst]: polynomia -> polynomials

File: docs/source/operations/transformations/horner.rst [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+
+Parameters
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+Required
+-------------------------------------------------------------------------------
+
+Below is a list of required parameters that can be set for the Horner polynomial
+transformation. As stated above, the transformation takes to forms, either using
+real or complex polynomia. These are divided into separate sections below.
```
Comment (mwtoews) [inline docs/source/operations/transformations/horner.rst]: polynomia -> polynomials

File: docs/source/operations/transformations/horner.rst [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+Optional
+-------------------------------------------------------------------------------
+
+.. option:: +range=<value>
+
+    Radius of the region of validity.
+
+.. option:: +uneg
+
+    Express latitude as southing. Only applies for complex polynomia.
```
Comment (mwtoews) [inline docs/source/operations/transformations/horner.rst]: polynomia -> polynomials

File: docs/source/operations/transformations/horner.rst [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+
+.. option:: +vneg
+
+    Express longitude as northing. Only applies for complex polynomia.
```
Comment (mwtoews) [inline docs/source/operations/transformations/horner.rst]: polynomia -> polynomials

File: docs/source/operations/transformations/horner.rst [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+
+.. option:: +fwd_origin=<northing,easting>
+
+    Coordinate of origin for the forward mapping
+
+.. option:: +inv_origin=<northing,easting>
+
+    Coordinate of origin for the inverse mapping
+
+Real polynomia
```
Comment (mwtoews) [inline docs/source/operations/transformations/horner.rst]: polynomia -> polynomials

File: docs/source/operations/transformations/horner.rst [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+    as described in :eq:`real_poly`.
+
+.. option:: +inv_v=<v_11,v_12,...,v_ij,..,v_mn>
+
+    Coefficients for the inverse transformation i.e. longitude to easting
+    as described in :eq:`real_poly`.
+
+
+
+Complex polynomia
```
Comment (mwtoews) [inline docs/source/operations/transformations/horner.rst]: polynomia -> polynomials

File: docs/source/operations/transformations/horner.rst [hunk 9]
```diff
@@ -0,0 +1,168 @@
...
+.. math::
+
+    N = 2d + 2
+
+.. option:: +fwd_c=<c_1,c_2,...,c_N>
+
+    Coefficients for the complex forward transformation
+    as described in :eq:`complex_poly`.
+
+.. option:: +inv_c=<coeffecient_1,coeffecient_2,...,coeffecient_n>
```
Comment (mwtoews) [inline docs/source/operations/transformations/horner.rst]: coeffecient -> coefficient (three times)

**Review by mwtoews — COMMENTED** [review]
Found a few typos (mostly 'polynomia') and other suggestions.

Perhaps also add a last section with a link to Wikipedia's article on [Horner's method](https://en.wikipedia.org/wiki/Horner%27s_method):
```
Further reading
################################################################################

#. `Wikipedia <https://en.wikipedia.org/wiki/Horner%27s_method>`_
```

File: docs/source/references.bib [hunk 10]
```diff
@@ -197,6 +197,20 @@ @TechReport{ONeilLaubscher1976
...
+    author = {A. C. Ruffhead},
+    title = {Introduction to multiple regression equations in datum transformations and their reversibility},
+    journal = {Survey Review},
+    volume = {50},
+    number = {358},
+    pages = {82-90},
+    year  = {2018},
+    publisher = {Taylor & Francis},
+    doi = {10.1080/00396265.2016.1244143},
+    URL = {https://doi.org/10.1080/00396265.2016.1244143},
```
[the remark this answers is not in the document]
Comment (kbevers) [reply docs/source/references.bib]: Good point. I just grabbed the bibtex citation from whatever place was trying to sell me the paper. I'll clean it up.

**Comment by kbevers** [comment]
> Found a few typos (mostly 'polynomia') and other suggestions.

Thanks. Somehow I have got into my head that that's the correct plural spelling but clearly that's not the case. I'll take care of the typos and your other suggestions later today.

**Comment by kbevers** [comment]
> yes, the current situation seems more logical to me. A 'pip install --upgrade sphinx' should probably fix your wrnings

It didn't:

```
(proj) C:\dev\proj\docs>make html
sphinx-build -b html -d build/doctrees  -W source build/html
Running Sphinx v1.8.1
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 207 source files that are out of date
updating environment: 207 added, 0 changed, 0 removed
C:\Users\b012349\AppData\Local\Continuum\anaconda3\envs\proj\lib\site-packages\sphinx\application.py:402: RemovedInSphinx20Warning: app.info() is now deprecated. Use sphinx.util.logging instead.
  RemovedInSphinx20Warning)
checking for C:\dev\proj\docs\source\references.bib in bibtex cache... not found
parsing bibtex file C:\dev\proj\docs\source\references.bib... parsed 28 entries

looking for now-outdated files... none found
pickling environment... done
checking consistency...
Warning, treated as error:
C:\dev\proj\docs\source\operations\options\R.rst:document isn't included in any toctree
Makefile:56: recipe for target 'html' failed
make: *** [html] Error 2

(proj) C:\dev\proj\docs>sphinx-build --version
sphinx-build 1.8.1
```

This is on Windows. Maybe there are differences across operating systems?

**Comment by kbevers** [comment]
I've added an exclude pattern to the sphinx config. That seems to be the lowest impact solution to the problem.

**Comment by rouault** [comment]
> I've added an exclude pattern to the sphinx config. That seems to be the lowest impact solution to the problem.

yes, actually the error message "R.rst:document isn't included in any toctree" is correct ! R.rst is only referenced through includes. Weird that this error isn't raised on Linux. Anyway, your fix looks great to me

**Comment by kbevers** [comment]
> yes, actually the error message "R.rst:document isn't included in any toctree" is correct ! 

Yeah, that's why I fixed it :-) I don't know why it isn't considered an error on linux or osx. Anyway, this should work for everybody.

**Pull request merged**